### ISSUE TITLE
SPK introspection backend needs to be deployable

### ISSUE THREAD

**Issue by samiyaakhtar** [microsoft/bedrock#1017]
**As a:** developer

**I want:** to run the backend and front end for the dashboard

**So that:** I can launch Spektate

**Describe the solution you'd like:**
As the work for https://github.com/microsoft/bedrock/issues/980 is almost complete, the following needs to be done before it's ready to merge:
- [x] New backend should be built as part of CI  
- [x] Create Dockerfile for backend and make it part of the pipeline
- [x] Write/Update documentation on how to start dashboard with these changes
~~-  `spk dashboard` command should use the new setup~~
~~- `spk` cli should consume the changes made to spektate package as part of https://github.com/microsoft/bedrock/issues/980~~

**Acceptance Criteria:**
Changes to spektate to bring in back end have been completed

**Describe alternatives you've considered:**
Consider the possibility of keeping the current implementation in place until this is finished

**Additional context:**
https://github.com/microsoft/bedrock/issues/938

**Does this require updates to documentation?:**
Yes

**Comment by andrebriggs** [comment]
Please split this into a smaller tasks
